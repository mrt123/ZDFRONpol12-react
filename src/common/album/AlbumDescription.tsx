import AlbumDescriptionEntry from "./AlbumDescriptionEntry";

const albumDescriptionStyles = {
  margin: "8px",
};

export interface AlbumDescriptionProps {
  title: string;
  artist: string;
  price?: number;
}

const AlbumDescription = ({ title, artist, price }: AlbumDescriptionProps) => {
  const priceToDisplay = price || "N/A";

  return (
    <div style={albumDescriptionStyles}>
      <AlbumDescriptionEntry value={title} />
      <AlbumDescriptionEntry value={artist} />
      <AlbumDescriptionEntry value={priceToDisplay} />
    </div>
  );
};

export default AlbumDescription;
