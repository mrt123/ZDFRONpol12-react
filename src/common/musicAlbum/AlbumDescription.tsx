import { daysAgo, daysAgoPretty } from "../dates";
import AlbumDescriptionEntry from "./AlbumDescriptionEntry";

const albumDescriptionStyles = {
  margin: "8px",
};

export interface AlbumDescriptionProps {
  title: string;
  artist: string;
  price?: number;
  releaseDate?: Date;
}

const AlbumDescription = ({
  title,
  artist,
  price,
  releaseDate,
}: AlbumDescriptionProps) => {
  let priceToDisplay;
  if (price) priceToDisplay = price;
  else priceToDisplay = "N/A";

  const daysAgoString = `released: ${daysAgoPretty(releaseDate)}`;

  return (
    <div style={albumDescriptionStyles}>
      <AlbumDescriptionEntry value={title} />
      <AlbumDescriptionEntry value={artist} />
      <AlbumDescriptionEntry value={priceToDisplay} />
      <AlbumDescriptionEntry value={daysAgoString} />
    </div>
  );
};

export default AlbumDescription;
