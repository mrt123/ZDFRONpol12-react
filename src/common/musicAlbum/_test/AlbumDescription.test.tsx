import { render } from "@testing-library/react";
import AlbumDescription from "../AlbumDescription";

test("AlbumDescription", () => {
  const { container } = render(
    <AlbumDescription title="Mam 8 motywacji!" artist="Kamil WG" />
  );
  expect(container).toMatchSnapshot();
});
