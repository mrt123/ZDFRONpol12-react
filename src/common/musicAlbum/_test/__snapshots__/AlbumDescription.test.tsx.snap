// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlbumDescription 1`] = `
<div>
  <div
    style="margin: 8px;"
  >
    <div
      style="margin: 5px; font-size: 11px;"
    >
      Mam 8 motywacji!
    </div>
    <div
      style="margin: 5px; font-size: 11px;"
    >
      Kamil WG
    </div>
    <div
      style="margin: 5px; font-size: 11px;"
    >
      N/A
    </div>
    <div
      style="margin: 5px; font-size: 11px;"
    >
      released: unknown
    </div>
  </div>
</div>
`;
